sklearn/neural_network/tests/test_mlp.py

```diff
@@ -965,3 +965,28 @@ def test_mlp_partial_fit_after_fit(MLPEstimator):
     msg = "partial_fit does not support early_stopping=True"
     with pytest.raises(ValueError, match=msg):
         mlp.partial_fit(X_iris, y_iris)
+
+
+def test_mlp_diverging_loss():
+    """Test that a diverging model does not raise errors when early stopping is enabled.
+
+    Non-regression test for:
+    https://github.com/scikit-learn/scikit-learn/issues/29504
+    """
+    mlp = MLPRegressor(
```

Comment:
We need to test both the regressor and the classifier.

Reply:
Because a classifier predicts 0s and 1s, the accuracy score will never have NaNs